pallets/circuit/src/lib.rs

```diff
@@ -1752,15 +1774,16 @@ impl<T: Config> Pallet<T> {
                         EscrowedBalanceOf<T, T::Escrowed>,
                     >,
                 ) -> SecurityLvl {
-                    fn is_escrowed<T: Config>(chain_id: &ChainId) -> bool {
-                        let gateway_type = <T as Config>::Xdns::get_gateway_type_unsafe(chain_id);
-                        gateway_type == GatewayType::ProgrammableInternal(0)
-                            || gateway_type == GatewayType::OnCircuit(0)
-                    }
-                    if is_escrowed::<T>(&side_effect.target) {
-                        return SecurityLvl::Escrowed
+                    let gateway_type =
```

Comment:
This is a bit problematic because escrow is only possible if the `security_coordinates` is set in xdns entry. Wouldn't it make sense to pass security level as a param and ensure that its possible (aka has escrow contract)

Reply:
> This is a bit problematic because escrow is only possible if the `security_coordinates` is set in xdns entry. Wouldn't it make sense to pass security level as a param and ensure that its possible (aka has escrow contract)

This is just removing the static function for a simple bool check and probably not for this PR. Could you write an issue for this?